src/libasr/alloc.h

```diff
@@ -52,15 +52,25 @@ class Allocator
         // force new_chunk() not to get inlined, but there is no standard way of
         // doing it. This try/catch approach effectively achieves the same using
         // standard C++.
+#ifdef LCOMPILERS_FAST_ALLOC
         try {
+#endif
             LCOMPILERS_ASSERT(start != nullptr);
             size_t addr = current_pos;
             current_pos += align(s);
-            if (size_current() > size_total()) throw std::bad_alloc();
+            if (size_current() > size_total()) {
+#ifdef LCOMPILERS_FAST_ALLOC
+                throw std::bad_alloc();
+#else
+                return new_chunk(s);
+#endif
+            }
             return (void*)addr;
+#ifdef LCOMPILERS_FAST_ALLOC
         } catch (const std::bad_alloc &e) {
             return new_chunk(s);
         }
+#endif
```

Comment:
I think, this can be reverted for now. We can add it later when building `lpython` in release mode.

Reply:
Sorry about this. LFortran needs it. Let's add it in LPython as well. (I assumed we could diverge here slightly and later make them same. Maybe its better to have equal LibASRs.)